Neos.ContentRepository.Core/Classes/Feature/NodeModification/Dto/SerializedPropertyValue.php

```diff
@@ -18,24 +18,40 @@
  * "Raw" / Serialized property value as saved in the event log // in projections.
  *
  * This means: "value" must be a simple PHP data type (no objects allowed!)
+ * Null is not permitted. To unset a property {@see UnsetPropertyValue} must be used.
  *
  * @api used as part of commands/events
  */
 final class SerializedPropertyValue implements \JsonSerializable
 {
     /**
-     * @param int|float|string|bool|array<int|string,mixed>|\ArrayObject<int|string,mixed>|null $value
+     * @param int|float|string|bool|array<int|string,mixed>|\ArrayObject<int|string,mixed> $value
      */
-    public function __construct(
-        public readonly int|float|string|bool|array|\ArrayObject|null $value,
+    private function __construct(
+        public readonly int|float|string|bool|array|\ArrayObject $value,
         public readonly string $type
     ) {
     }
 
+    /**
+     * If the value is NULL an unset-property instruction will be returned instead.
+     *
+     * @param int|float|string|bool|array<int|string,mixed>|\ArrayObject<int|string,mixed>|null $value
+     */
+    public static function create(
+        int|float|string|bool|array|\ArrayObject|null $value,
+        string $type
+    ): self|UnsetPropertyValue {
+        if ($value === null) {
+            return UnsetPropertyValue::get();
+        }
+        return new self($value, $type);
+    }
```

Comment:
Do we need to expose the "null"-hack to the new constructor?
I would suggest to change this to:
```suggestion
    public static function create(
        int|float|string|bool|array|\ArrayObject $value,
        string $type
    ): self|UnsetPropertyValue {
        return new self($value, $type);
    }
```

and then...